dateutil/test/test_parser.py

```diff
@@ -748,6 +747,11 @@ def testUnspecifiedDayFallbackFebLeapYear(self):
         self.assertEqual(parse("Feb 2008", default=datetime(2010, 1, 31)),
                          datetime(2008, 2, 29))
 
+    def testTzinfoCouldbeNone(self):
```

Comment:
Can you also add a separate test that checks that `lambda *args: None` also works without raising an error?